statsd/validator.go

```diff
@@ -24,3 +25,13 @@ func CheckName(stat string) error {
 	}
 	return nil
 }
+
+func CheckIsIP(hostport string) (bool, error) {
```

Comment:
I think it's better if this function is private, and does not return an error, and assumes false if net.SplitHostPort returns an error (let resolution happen like normally during dial)

Reply:
My thinking is there could be a situation where the go library for some unknown reason cannot definitively Split a host/port, but still may be able to dial somehow.

Reply:
If you do that, I would rename the function from `CheckIsIP` to `absolutelyIsIP`

Reply:
Nice idea. 👍